src/restconfig/src/test/java/org/geoserver/rest/catalog/CoverageStoreFileUploadTest.java

```diff
@@ -576,4 +572,108 @@ private void configureCoverageInfo(
         // add the store
         getCatalog().add(cinfo);
     }
+
+    @Test
+    public void testDefaultBehaviourUpdateBBoxPOST() throws Exception {
+        setUpBBoxTest("bboxtest");
+        byte[] bytes = null;
+        CoverageStoreInfo storeInfo = getCatalog().getCoverageStoreByName("bboxtest");
+        assertNotNull(storeInfo);
+        CoverageInfo ci = getCatalog().getCoverageByName("bboxtest");
+        assertNotNull(ci);
+        assertEquals(storeInfo, ci.getStore());
+        // Harvesting
+        URL zipHarvest = getClass().getResource("test_bbox_granules.zip");
+        try (InputStream is = zipHarvest.openStream()) {
+            bytes = IOUtils.toByteArray(is);
+        }
+        // Create the POST request
+        MockHttpServletRequest request =
+                createRequest(
+                        RestBaseController.ROOT_PATH
+                                + "/workspaces/gs/coveragestores/bboxtest/file.imagemosaic");
+        request.setMethod("POST");
+        request.setContentType("application/zip");
+        request.setContent(bytes);
+        request.addHeader("Content-type", "application/zip");
+        dispatch(request);
+        testBBoxLayerConfiguration(
+                storeInfo, (current, old) -> assertNotEquals(current, old), catalog);
+        CoverageInfo coverage = getCatalog().getResourceByName("bboxtest", CoverageInfo.class);
+        if (coverage != null) {
+            removeStore(
+                    coverage.getStore().getWorkspace().getName(), coverage.getStore().getName());
+        }
+    }
+
+    @Test
+    public void testUpdateBBoxTrueParameterPOST() throws Exception {
```

Comment:
Good to see tests for harvest and re-harvest. The delete path was modified too, is there a test for it?

Reply:
yes those are in the StructuredCoverageStoresTest class https://github.com/geoserver/geoserver/pull/4156/commits/9c3df0dec3079791dd5f99101180e5ffe5e92bb9#diff-12372cf7938b4aed6e65eba2d824237aR871